File: backend/src/main/java/com/logistics/entity/OrderProduct.java
package com.logistics.entity;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Entity
@Table(name = "order_products")
@Data
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class OrderProduct {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "orderId", nullable = false)
    private Order order;

    @ManyToOne
    @JoinColumn(name = "productId", nullable = false)
    private Product product;

    @Column(nullable = false)
    private Integer quantity = 1;

    @Column(nullable = false)
    private Integer price;

    @Column(nullable = false)
    private Integer deliveredQuantity = 0;

    @Column(nullable = false)
    private Integer returnedQuantity = 0;

    @Version
    @Column(nullable = false, columnDefinition = "BIGINT NOT NULL DEFAULT 0")
    private Long version = 0L;

    @CreatedDate
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    private LocalDateTime updatedAt;
}

File: frontend/src/types/orderProduct.ts
export interface OrderProduct {
  id?: number;
  productId: number;
  productPrice: number;
  productType: string;
  productStock: number;
  productCode: string;
  productName: string;
  productWeight: number;
  quantity: number;
  price: number;
  deliveredQuantity?: number;
  returnedQuantity?: number;
}

export interface OrderProductPrint {
  productName: string;
  quantity: number;
}

export interface OrderProductRequest {
  productId: number;
  quantity: number;
}